cli/info.rs

```diff
@@ -1,404 +1,232 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
 
 use crate::colors;
-use crate::global_state::GlobalState;
-use crate::module_graph::{ModuleGraph, ModuleGraphFile, ModuleGraphLoader};
+use crate::MediaType;
 use crate::ModuleSpecifier;
-use crate::Permissions;
-use deno_core::error::AnyError;
-use serde::ser::Serializer;
-use serde::Serialize;
-use std::collections::{BTreeMap, HashMap, HashSet};
-use std::sync::Arc;
 
-// TODO(bartlomieju): rename
-/// Struct containing a module's dependency information.
-#[derive(Serialize)]
-#[serde(rename_all = "camelCase")]
-pub struct ModuleDepInfo {
-  module: String,
-  local: String,
-  file_type: String,
-  compiled: Option<String>,
-  map: Option<String>,
-  dep_count: usize,
-  #[serde(skip_serializing)]
-  deps: FileInfoDepTree,
-  total_size: Option<usize>,
-  files: FileInfoDepFlatGraph,
+use serde::ser::SerializeSeq;
+use serde::Serialize;
+use serde::Serializer;
+use std::cmp::Ordering;
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::fmt;
+use std::path::PathBuf;
+
+/// Serialize a `MediaType` enum into a human readable string.  The default
+/// serialization for media types is and integer.
+fn serialize_media_type<S>(mt: &MediaType, s: S) -> Result<S::Ok, S::Error>
+where
+  S: Serializer,
+{
+  s.serialize_str(&format!("{}", mt))
```

Comment:
Presumably we want to do this everywhere we serialize MediaType. Maybe `impl Serialize for MediaType` where it's defined?

Reply:
No, because we have a serialize of `MediaType` that goes to the equivalent of the enum in JavaScript for `tsc`.  Though in the new infrastructure we don't serailize that at all to send into JavaScript.

I could refactor it so this is the default serialization and we modify the `module_graph` (legacy) to use a custom serializer, but I would just rather address it once we have completed the migration.